docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,280 @@
+
+Once the `ShootedSeed` resource and its controller are considered sufficiently stable, the current `use-as-seed` annotation and the controller mentioned above should be marked as deprecated and eventually removed.
+
+**Note:** Bootstrapping a new seed might require additional provider-specific actions to the ones performed automatically by the shooted seed controller. For example, on Azure this might include getting a new subscription, extending quotas, etc. This could eventually be automated by introducing an extension mechanism for the Gardener seed bootstrapping flow, to be handled by a new type of controller in the provider extensions. However, such an extension mechanism is not in the scope of this proposal and might require a separate GEP.
+
+### ShootedSeedSets
+
+Similarly to a [ReplicaSet](https://kubernetes.io/docs/concepts/workloads/controllers/replicaset/), the purpose of a ShootedSeedSet is to maintain a stable set of replica [ShootedSeeds](#shootedseeds) available at any given time. As such, it is used to guarantee the availability of a specified number of identical ShootedSeeds.
+
+#### ShootedSeedSet Resource
+
+The `ShootedSeedSet` resource has a `selector` field that specifies how to identify ShootedSeeds it can acquire, a number of `replicas` indicating how many ShootedSeeds it should be maintaining, and a shooted seed `template` specifying the data of new ShootedSeeds it should create to meet the number of replicas criteria. A ShootedSeedSet then fulfills its purpose by creating and deleting ShootedSeeds as needed to reach the desired number. 
+
+A ShootedSeedSet is linked to its ShootedSeeds via the ShootedSeeds' `metadata.ownerReferences` field, which specifies what resource the current object is owned by. All ShootedSeeds acquired by a ShootedSeedSet have their owning ShootedSeedSet's identifying information within their `ownerReferences` field. 
+
+Example `ShootedSeedSet` resource:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: ShootedSeedSet
+metadata:
+  name: crazy-botany
+  namespace: garden
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      foo: bar
+  template:
+    metadata:
+      labels:
+        foo: bar
+    spec: # ShootedSeed resource
+      shootTemplate: 
+        ...
+      seedTemplate: 
+        ...
+      gardenlet: 
+        ...
+```
+
+#### ShootedSeedSet Controller
+
+ShootedSeedSets are reconciled by a new *shooted seed set controller* in `gardenlet`. During the reconciliation this controller creates and deletes ShootedSeeds in response to changes to the `replicas` and `selector` fields.
```

Comment:
Hm, generally we have 1 replica of kube-controller-manager that watches on ReplicaSet and maintains the corresponding Pods. If we have ShootedSeedSet and multiple gardenlets, how looks like the flow? A `gardenlet` is watching for all `ShootedSeedSet`s ?

Reply:
As I mentioned above, I completely rewrote the ShootedSeed section to reflect learnings from a PoC I am currently working on. The major change is that ShootedSeeds are actually reconciled by a controller in GCM, not `gardenlet` (which still delegates a lot of the work to `gardenlet` by creating a `SeedRegistration` resource, see the updated proposal). With this change, the issue you mention above is solved naturally by moving the *shooted seed set controller* also to GCM. The controller would produce multiple ShootedSeeds, which are then reconciled by another GCM controller to produce multiple Shoots and SeedRegistrations, which are then reconciled by controllers in `gardenlet`.